Services/DiscountsCalculator.cs

```diff
@@ -49,19 +49,32 @@ public double CalculateUniversalDiscount(Product product)
             double PostTaxDiscount = 0;
             if (storeRules.universalDiscount?.Type == DiscountType.POST_TAX)
             {
-                PostTaxDiscount = Math.Round(PostTaxDiscount + CalculateUniversalDiscount(product), 2);
+                PostTaxDiscount = Math.Round(PostTaxDiscount + CalculateUniversalDiscount(product.Price), 2);
             }
             if (storeRules.specialDiscount?.Type == DiscountType.POST_TAX)
             {
-                PostTaxDiscount = Math.Round(PostTaxDiscount + CalculateSpecialDiscount(product), 2);
+                PostTaxDiscount = Math.Round(PostTaxDiscount + CalculateSpecialDiscount(product, product.Price), 2);

```

Comment:
you should calculate the post tax discount based on the price after pre tax discount
but you're using the original price